spark-indexer/src/main/java/org/uniprot/store/spark/indexer/proteome/ProteomeRDDReader.java

```diff
@@ -64,11 +102,126 @@ private Dataset<Row> loadRawXml() {
         return data;
     }
 
-    private StructType geProteomeXMLSchema() {
+    public static StructType geProteomeXMLSchema() {
```

Comment:
Use XSD file to create the schema. You can get latest schema from https://gitlab.ebi.ac.uk/uniprot/framework/unp.fw.proteomes/-/raw/master/proteomes-xml/src/main/resources/META-INF/xml/proteome.xsd

Reply:
It should be getProteomeXMLSchema. See this method's usage and check if we can pass the XSD file directly to spark.